File: examples/Python/iss_detector.py
#!/usr/bin/env python3
# @file      iss_detector.py
# @author    Ignacio Vizzo     [ivizzo@uni-bonn.de]
#
# Copyright (c) 2020 Ignacio Vizzo, all rights reserved
import open3d_keypoints as o3d


def main():
    bunny = o3d.io.read_triangle_mesh('../TestData/Bunny.ply')
    bunny_cloud = bunny.sample_points_uniformly(len(bunny.vertices))
    iss_keypoints = o3d.keypoints.compute_iss_keypoints(bunny_cloud)
    o3d.visualization.draw_geometries([iss_keypoints, bunny_cloud])


if __name__ == "__main__":
    main()

File: src/Python/open3d_pybind/keypoints/iss_detector.cpp
// @file      iss_detector.cpp
// @author    Ignacio Vizzo     [ivizzo@uni-bonn.de]
//
// Copyright (c) 2020 Ignacio Vizzo, all rights reserved
#include "Open3D/Geometry/PointCloud.h"
#include "Open3D/Keypoints/ISSKeypointDetector.h"
#include "open3d_pybind/docstring.h"
#include "open3d_pybind/keypoints/keypoints.h"

namespace open3d {

void pybind_iss_detector(py::module &m) {
    m.def("compute_iss_keypoints", &keypoints::ComputeISSKeypoints,
          "Function to compute ISS keypoints for a point cloud", "input"_a);
    docstring::FunctionDocInject(m, "compute_iss_keypoints",
                                 {{"input", "The Input point cloud."}});
}

}  // namespace open3d
